pkg/validation/app.go

```diff
@@ -64,6 +65,12 @@ func (v *Validator) validateCatalog(ctx context.Context, cr v1alpha1.App) error
 }
 
 func (v *Validator) validateConfig(ctx context.Context, cr v1alpha1.App) error {
+	_, hasManagedConfig := cr.Annotations[annotation.ConfigMajorVersion]
+	if hasManagedConfig && (key.AppConfigMapName(cr) == "" || key.AppSecretName(cr) == "") {
+		// wait for config-controller setting app CR configmap and secret
+		return microerror.Mask(appDependencyNotReadyError)
```

Comment:
```suggestion
		return microerror.Maskf(appDependencyNotReadyError, "user ConfigMap or Secret not set")
```

Reply:
But it is not `user` configmap or secret. I'll set then `ConfigMap or Secret not set`

Reply:
Edited.